test/fake-ws.js

```diff
@@ -101,75 +97,44 @@ WebSocket.prototype.abort = function () {
+        ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.RESULT, WAMP_MSG_SPEC.INVOCATION].includes(send_data?.data?.[0]))
+    ) {
+        if ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.INVOCATION].includes(send_data.data[0])) {
+            options = send_data.data[3];
+        } else if ([WAMP_MSG_SPEC.RESULT].includes(send_data.data[0])) {
+            options = send_data.data[2];
         }
 
-        // Check for PPT mode and encode payload with appreciate serializer
-        if (opts.ppt_scheme &&
-            opts.ppt_serializer &&
-            opts.ppt_serializer !== 'native') {
-            switch (opts.ppt_serializer) {
-                case 'cbor':
-                    pptSerializer = new CborSerializer();
-                    break;
-                case 'msgpack':
-                    pptSerializer = new MsgpackSerializer();
-                    break;
-                case 'json':
-                    pptSerializer = new JsonSerializer();
-                    break;
-            }
+        // Check for PPT mode and encode payload with serializer
+        if (options.ppt_scheme &&
+            options.ppt_serializer &&
+            options.ppt_serializer !== 'native') {
+            pptSerializer = serializers[options.ppt_serializer];
+
+            if ([WAMP_MSG_SPEC.EVENT, WAMP_MSG_SPEC.INVOCATION].includes(send_data.data[0])) {
+                const payload = pptSerializer.encode(send_data.data[4][0]);
+                const ruinedPayload = String(payload);
 
-            if (send_data.data[0] === WAMP_MSG_SPEC.EVENT) {
-                send_data.data[4] = [
-                    send_data.ruinPayload ?
-                        pptSerializer.encode(send_data.data[4][0]) + '123' :
-                        pptSerializer.encode(send_data.data[4][0])
-                ];
-            } else if (send_data.data[0] === WAMP_MSG_SPEC.RESULT) {
-                send_data.data[3] = [
-                    send_data.ruinPayload ?
-                        pptSerializer.encode(send_data.data[3][0]) + '123' :
-                        pptSerializer.encode(send_data.data[3][0])
-                ];
-            } else if (send_data.data[0] === WAMP_MSG_SPEC.INVOCATION) {
-                send_data.data[4] = [
-                    send_data.ruinPayload ?
-                        pptSerializer.encode(send_data.data[4][0]) + '123' :
-                        pptSerializer.encode(send_data.data[4][0])
-                ];
+                send_data.data[4] = [send_data.ruinPayload ? ruinedPayload : payload];
+            } else if ([WAMP_MSG_SPEC.RESULT].includes(send_data.data[0])) {
+                const payload = pptSerializer.encode(send_data.data[3][0]);
+                const ruinedPayload = String(payload);
+
+                send_data.data[3] = [send_data.ruinPayload ? ruinedPayload : payload];
             }
         }
     }
 
-    // console.log('Data received by server:', rec_data);
```

Comment:
You removed all the commented-out logs. 
Well, sometimes when I add new tests and need to debug them, I use it.  That's why I am keeping them... 
That's because it is a bit tricky sometimes to add a correct wamp fake server message response on the first try.
We can rearrange them somehow for clarity but let's not remove that totally.

Reply:
And yeah) Sometimes console.log is better tool then step-by-step debugger :)

Reply:
Oh, I'm sorry I thought they were just dead code haha
What do you think of adding them under a "isDebugMode" flag so they blend better with the rest of the code?

Reply:
👍🏻  Great idea!